### PR TITLE
Birth of the KD Tree Accel

### PR DESCRIPTION
**Pull request by cameroncuster**
Not looking to merge, just putting this up so everyone can take a look at the sauce I'm cooking with + give feedback

### CODE REVIEW

**Comment by tylerhm** [comment]
IT HATH BEEN BORN 🙏🏼

File: lib/accelerators/kd_tree.hpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#define CPPTRACE_LIB_HITTABLE_KD_TREE_HPP_
+
+#include "../common/common.hpp"
+#include "../common/ray.hpp"
+#include "../hittable/bounding/aabb.hpp"
+#include "../hittable/hit.hpp"
+#include "../hittable/hittable.hpp"
+#include "../hittable/hittable_list.hpp"
+
+struct KDTree : Hittable {
```
Comment (TylerMathis) [inline lib/accelerators/kd_tree.hpp:15]: `struct KDTree : public Hittable {`

Comment (cameroncuster) [inline lib/accelerators/kd_tree.hpp]: Why should it be different than what BVH has?

File: lib/accelerators/kd_tree.hpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+#ifndef CPPTRACE_LIB_HITTABLE_KD_TREE_HPP_
+#define CPPTRACE_LIB_HITTABLE_KD_TREE_HPP_
+
+#include "../common/common.hpp"
+#include "../common/ray.hpp"
+#include "../hittable/bounding/aabb.hpp"
+#include "../hittable/hit.hpp"
+#include "../hittable/hittable.hpp"
+#include "../hittable/hittable_list.hpp"
+
```
Comment (tylerhm) [inline lib/accelerators/kd_tree.hpp:14]: Do we want to explicitly include the standard libraries we are using here? Idk best practice

```cpp
#include <memory>
#include <vector>
etc
```

File: lib/accelerators/kd_tree.hpp (hunk 1 above)
Comment (tylerhm) [inline lib/accelerators/kd_tree.hpp]:
```cpp
struct KDTree : public Hittable {
```

**Review by tylerhm — CHANGES_REQUESTED** [review]
small things

File: lib/accelerators/kd_tree.hpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  KDTree() = default;
+
+  KDTree(const HittableList &);
+
+  KDTree(const std::vector<std::shared_ptr<Hittable>> &);
+
+  int buildTree(const std::vector<int>, AABB);
+
+  bool hit(const Ray &, Hit &, const double, const double) const override;
+};
```
Comment (tylerhm) [inline lib/accelerators/kd_tree.hpp:38]: Need bounding box

**Review by tylerhm — CHANGES_REQUESTED** [review]

**Review by tylerhm — APPROVED** [review]

**Pull request merged**